app/models/product.rb

```diff
@@ -6,6 +6,19 @@ class Product < ApplicationRecord
   has_many :orderDetails
   has_many :rates
 
+  validates :name,presence: true, length: {maximum: 100}
+  validates :content,presence: true
+  VALID_PRICE_REGEX = /\A(\$)?(\d+)(\.|,)?\d{0,2}?\z/
+  validates :price,presence: true,format: {with: VALID_PRICE_REGEX },uniqueness: { case_sensitive: false }
+  mount_uploader :img_detail, ImageUploader
+  validate :img_size
+  private
+  def img_size
+    if img_detail.size > 5.megabytes
+      errors.add(:img_detail, "should be less than 5MB")
+    end
+  end
+
   def self.search search
```

Comment:
- chỗ này viết thành `scope` thay cho `method` có được ko em?
- mà method `filtersearch` em đang xử lý logic nhiều quá, em tách ra thành một service nhé. Em có thể tham khảo thêm ở đây:
https://www.engineyard.com/blog/keeping-your-rails-controllers-dry-with-services
https://hackernoon.com/service-objects-in-ruby-on-rails-and-you-79ca8a1c946e